### ISSUE TITLE
Page.update(16) works different in 2020.1 compared to 2020.0 and previous versions

### ISSUE THREAD

**Issue by maxbrown13** [SketchUp/api-issue-tracker#468]
Using SU 2020.1 and Win10.

Using ruby code to hide groups on multiple pages at once. 
Page.update(16) doesn't work as before. Now have to use Page.update(384)
Expected; groups remain hidden after hiding groups on a page, update-ing the page and switching to another page and back.

Steps to test:
1. create a group
2. create 2 pages
3. select the group
4. run ruby code
5. switch pages - object again visible
6 change 16 to 384 in code
7. switch pages - object remains hidden

```ruby
# Complete reprodusible code snippet. (Important!)
```
model = Sketchup.active_model
cur_page = model.pages.selected_page
sel = model.selection
sel[0].hidden=true
status = cur_page.update(16)
```c
// Complete reprodusible code snippet. (Important!)
```

**Comment by Eneroth3** [comment]
Reference to forum post: https://forums.sketchup.com/t/ruby-bug-or-documentation-correction-su-2020-1-page-update-16-vs-page-update-384/124424

My proposed fix to have old extensions be as compatible as possible in SU 2020 is that 16 internally gets treated as `128 & 256`.

**Comment by DanRathbun** [comment]
> My proposed fix to have old extensions be as compatible as possible in SU 2020 is that 16 internally gets treated as `128 & 256`.

Except this would be different behavior and be a breaking change for old extensions. 
Ie, in previous versions only **toplevel** geometry and objects were saved.

Other places in the API the old API has been deprecated but actually and **does not work correctly** in 2020.1+.

Example [`Sketchup::Page#use_hidden?`](https://ruby.sketchup.com/Sketchup/Page.html#use_hidden%3F-instance_method) and [`Sketchup::Page#use_hidden=`](https://ruby.sketchup.com/Sketchup/Page.html#use_hidden=-instance_method)

So I can understand since these two above listed API features are broken, that [`Sketchup::Page#update(16)`](https://ruby.sketchup.com/Sketchup/Page.html#update-instance_method) could also be broken.

In other words, the way that the new nested "hidies" are implemented probably makes it so that there is no way to be fully backward compatible.

I was thinking perhaps that [`Sketchup::Page#update(16)`](https://ruby.sketchup.com/Sketchup/Page.html#update-instance_method)should continue to ONLY "remember" toplevel geometry and objects, in order to be backward compatible.

But IF the new implementation cannot do this, what is the choice?
* 16 is deprecated and becomes a no-op.
* 16 is "rewired" as Julia suggests even if it gives newer behavior.

@thomthom ?

**Comment by Eneroth3** [comment]
It would be a different behavior but much closer than not updating anything at all. If an extension changes visibility and call this method, the chance that the user has also changed the visibility in nested contexts and have it accidentally updated exists but is not a deal breaker.

That said, it would also be possible to restore the previous behavior exactly as it was, and have 16 only update objects and geometry at the top drawing context for the scene, but would require special handling in the back end.

**Comment by DanRathbun** [comment]
I would not mind if they did as you suggest, (ie 16 does both `128` & `256`,) but they could also fix the two broken methods to set and get both geometry and objects at the same time.

Ie ... [`Sketchup::Page#use_hidden?`](https://ruby.sketchup.com/Sketchup/Page.html#use_hidden%3F-instance_method) could be fixed to return `true` only if **both** geometry and object hidden flags were true.

**Comment by thomthom** [comment]
I'm not clear on the exact details, I need to double check with the team, but this was a change in the core of SketchUp on how scenes and hidden geometry worked which didn't leave room to remain 100% API compatible.

Btw, I noticed the docs  for the `#update` method doesn't mention the constants for the numeric values;

```
PAGE_USE_CAMERA
PAGE_USE_RENDERING_OPTIONS
PAGE_USE_SHADOWINFO
PAGE_USE_SKETCHCS
PAGE_USE_HIDDEN
PAGE_USE_HIDDEN_GEOMETRY
PAGE_USE_HIDDEN_OBJECTS
PAGE_USE_LAYER_VISIBILITY
PAGE_USE_SECTION_PLANES
PAGE_USE_ALL 
```

They are hidden in the list of top level constants; https://ruby.sketchup.com/top-level-namespace.html

I'll get an issue logged for that as well. Magic numbers are no good.

> Other places in the API the old API has been deprecated but actually and does not work correctly in 2020.1+.
> 
> Example Sketchup::Page#use_hidden? and Sketchup::Page#use_hidden=

Use hidden was split into two different properties, it was no way for them to work exactly as they did. However, I think the intent was for `#use_hidden=` to set both of the new properties. And I don't recall if `#use_hidden?` was set to an AND or OR comparison of them.

**Comment by DanRathbun** [comment]
> They are hidden in the list of top level constants;

```ruby
Sketchup::Page::USE_CAMERA ||= PAGE_USE_CAMERA
Sketchup::Page::USE_RENDERING_OPTIONS ||= PAGE_USE_RENDERING_OPTIONS
Sketchup::Page::USE_SHADOWINFO ||= PAGE_USE_SHADOWINFO
Sketchup::Page::USE_SKETCHCS ||= PAGE_USE_SKETCHCS
Sketchup::Page::USE_HIDDEN ||= PAGE_USE_HIDDEN
Sketchup::Page::USE_HIDDEN_GEOMETRY ||= PAGE_USE_HIDDEN_GEOMETRY
Sketchup::Page::USE_HIDDEN_OBJECTS ||= PAGE_USE_HIDDEN_OBJECTS
Sketchup::Page::USE_LAYER_VISIBILITY ||= PAGE_USE_LAYER_VISIBILITY
Sketchup::Page::USE_SECTION_PLANES ||= PAGE_USE_SECTION_PLANES
Sketchup::Page::USE_ALL ||= PAGE_USE_ALL
```

:?:

> However, I think the intent was for `#use_hidden=` to set both of the new properties. And I don't recall if `#use_hidden?` was set to an AND or OR comparison of them.

But the docs don't say this ... but should if it;s true. I think I showed by my report "_elsewhere_" that `#use_hidden=` on 2020.1 would confuse `#use_hidden?` and the two could get out of sync, especially with what the 2 new query methods think are the state of things.

> Use hidden was split into two different properties, it was no way for them to work exactly as they did.

I guessed as much (above.)

> I'm not clear on the exact details, I need to double check with the team, but this was a change in the core of SketchUp on how scenes and hidden geometry worked which didn't leave room to remain 100% API compatible.

Please do, and then please add to the docs (instead of just marking the 3 old methods as deprecated.) Coders need to realize what will happen in 2020.1+ when they call `page.update(16)` from old extensions.
And what would happen in older SketchUp version when a bitmask higher than `64` is used to update a page. (I assume since it's a bitmask the **new** high order bits are ignored.)
